vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1205,16 +1206,51 @@ def _get_prompt_logprobs_dict(
             prompt_token_ids = torch.tensor(request.prompt_token_ids).to(
                 self.device, non_blocking=True)
 
+            # Set up target LogprobsTensors object.
+            logprobs_tensors = in_progress_dict.get(req_id)
+            if not logprobs_tensors:
+                # Create empty logprobs CPU tensors for the entire prompt.
+                # If chunked, we'll copy in slice by slice.
+                logprob_token_ids = torch.empty(
+                    (num_prompt_tokens - 1, num_prompt_logprobs + 1),
+                    dtype=torch.int32,
+                    device="cpu")
+
+                logprobs_tensors = LogprobsTensors(
+                    logprob_token_ids=logprob_token_ids,
+                    logprobs=torch.empty_like(logprob_token_ids,
+                                              dtype=torch.float32),
+                    selected_token_ranks=torch.empty(num_prompt_tokens - 1,
+                                                     dtype=torch.int32,
+                                                     device="cpu"),
+                )
+
             # Determine number of logits to retrieve.
-            start_tok = request.num_computed_tokens + 1
+            start_idx = request.num_computed_tokens
+            start_tok = start_idx + 1
             num_remaining_tokens = num_prompt_tokens - start_tok
-            if num_tokens < num_remaining_tokens:
+            if num_tokens <= num_remaining_tokens:
                 # This is a chunk, more tokens remain.
+                # In the == case, there are no more prompt logprobs to produce
+                # but we want to defer returning them to the next step where we
+                # have new generated tokens to return.
                 num_logits = num_tokens
+                if start_idx == 0:
```

Comment:
We can avoid special casing `if start_idx == 0` and `start_idx != 0` if we:

- move `in_progress_dict[req_id] = logprobs_tensors` under the `if not logprobs_tensors` branch above
- move `del in_progress_dict[req_id]` under the `for req_id in completed_prefill_reqs:` loop below

This will just make the code easier to follow at the expense of slightly more work. The design goal of this is "simple" over "optimal"

Reply:
sure, I guess I was just avoiding adding/removing in the common non-chunked case but you're probably right that simpler code is a better option

Reply:
yeah, its a small nit but i think making this as simple as possible is a good idea